core/src/impl/Kokkos_HostThreadTeam.hpp

```diff
@@ -848,6 +1147,36 @@ Impl::TeamThreadRangeBoundariesStruct<iType,Impl::HostThreadTeamMember<Space> >
   loop_boundaries.thread.team_reduce( reducer );
 }*/
 
+template <Kokkos::Iterate Direction, size_t Rank, typename iType,
+          typename Member, typename Closure, typename Reducer>
+KOKKOS_INLINE_FUNCTION
+    typename std::enable_if_t<Kokkos::is_reducer<Reducer>::value &&
+                              Impl::is_host_thread_team_member<Member>::value>
+    parallel_reduce(Impl::MDTeamThreadRangeBoundariesStruct<
+                        Direction, Rank, iType, Member> const& boundaries,
+                    Closure const& closure, Reducer const& reducer) {
+  typename Reducer::value_type value;
+  reducer.init(value);
+
+  parallel_for(boundaries, [&](auto... is) { closure(is..., value); });
+
+  boundaries.thread.team_reduce(reducer, value);
+}
+
+template <Kokkos::Iterate Direction, size_t Rank, typename iType,
+          typename Member, typename Closure, typename ValueType>
+KOKKOS_INLINE_FUNCTION
+    std::enable_if_t<!Kokkos::is_reducer<ValueType>::value &&
+                     Impl::is_host_thread_team_member<Member>::value>
+    parallel_reduce(Impl::MDTeamThreadRangeBoundariesStruct<
+                        Direction, Rank, iType, Member> const& boundaries,
+                    Closure const& closure, ValueType& result) {
+  Sum<ValueType> reducer(result);
+  reducer.init(result);
+
+  parallel_reduce(boundaries, closure, reducer);
```

Comment:
For the MD versions of Team functions, we referred to their non-MD counterparts and kept implementations close to them, so that we don't introduce any behavior that wasn't supported by the non-MD Team functions.

`parallel_reduce` that takes in a `Reducer` follows the non-MD `parallel_reduce`, which also uses `parallel_for`. We found an opportunity to have `parallel_reduce` that takes in a `ValueType` to simply call through the MD version of `parallel_reduce`, because their implementations were essentially the same in their non-MD versions.